File: Leetcode/590_N-ary_Tree_Postorder_Traversal.cpp
/*
// Definition for a Node.
class Node {
public:
    int val;
    vector<Node*> children;

    Node() {}

    Node(int _val, vector<Node*> _children) {
        val = _val;
        children = _children;
    }
};
*/
class Solution {
public:
    vector<int> ans;
    vector<int> postorder(Node* root) {
        if(root != nullptr)
        {
            for(auto t: root->children)
            {
                postorder(t);
            }
            ans.push_back(root->val);
        }
        return ans;
    }
};

File: Leetcode/0590/590_N-ary_Tree_Postorder_Traversal.js
/**
 * // Definition for a Node.
 * function Node(val,children) {
 *    this.val = val;
 *    this.children = children;
 * };
 */
/**
 * @param {Node} root
 * @return {number[]}
 */
var postorder = function (root, result = []) {
    if (root) {
        root.children.forEach(child => postorder(child, result));

        result.push(root.val);

        return result;
    }

    return [];
};